sourcetracker/_sourcetracker.py

```diff
@@ -9,278 +9,277 @@
 # ----------------------------------------------------------------------------
 from __future__ import division
 
-import os
-from copy import copy
 import numpy as np
 from skbio.stats import subsample_counts
 import pandas as pd
 from functools import partial
-from biom.table import Table
-import glob
-from tempfile import TemporaryDirectory
 
 
-def parse_mapping_file(mf_lines):
-    """Parse lines from a mapping file into a nested dictionary.
-
-    This function copies qiime.parse_mapping_file_to_dict, but does _not_ allow
-    commented lines at the top of a mapping file.
+def check_and_correct_data(feature_table, apply_fractional_value_correction):
+    '''Check and correct problems in the `feature_table`.
 
     Parameters
     ----------
-    mf_lines : list
-        Each entry of the list is a tab delimited string corresponding to a row
-        of the mapping file.
+    feature_table : pd.DataFrame
+        Feature table where rows are features and columns are samples.
+    apply_fractional_value_correction : boolean
+        Fractional counts cause errors in indexing. Uses np.ceil to remove
+        them if True.
 
     Returns
     -------
-    results : dict
-    """
-    headers = mf_lines[0].strip().split('\t')[1:]
-    results = {}
-    for line in mf_lines[1:]:
-        tmp = line.strip().split('\t')
-        results[tmp[0]] = {k: v.strip() for k, v in zip(headers, tmp[1:])}
-    return results
+    feature_table : pd.DataFrame
+        Corrected feature table if changes have been made or original table if
+        no changes occurred. Will automatically be cast as dtype=np.int64
+
+    Raises
+    ------
+    ValueError
+        If data contains values that Gibb's sampling cannot handle.
+    '''
+    acceptable_types = (int, float, np.int64, np.int32, np.float64, np.float32)
+    if not all(np.in1d(feature_table.dtypes, acceptable_types)):
+        raise TypeError('Feature table contains one or more columns which are '
+                        'not numeric type. This is likely due to boolean, '
+                        'string, or other data types being present. Data must '
+                        'contain exclusively real-valued integers or floats.')
+
+    if feature_table.isnull().any().any():
```

Comment:
What's with the double ``any`` here? Is there a simpler way to express that (e.g., using ``np.in1d``?)